### PR TITLE
Fix errors upload_max_filesize, wrong mimetypes for SonataAdmin

### PR DESCRIPTION
**Pull request by antoinemineau**
Hi.

I found a way to fix the issue when you upload a file bigger than the upload_max_filesize, and also on invalid mimetypes.

I added an exception if the clientsize is at  zero bytes, i'm guessing that it's not uploaded because of upload_max_filesize value. I had to set a providerReference because there is a validation on this field: can't be null or blank  (if there is no providerReference set, on top of the form it displays: this field cant be null, this field cant be blank).  I am catching this exception in the BaseProvider. I added in the validate method an error violation for the client size at zero bytes. The wrong mimetypes already had an error violation related, the issue was exception not being caught. I had to modify the last if statement so when the file isn't uploaded there is no check on the non existing file mimetypes.

Might not be the prettiest way to do it, but there is no more 500 errors.

Screenshot of file bigger than the upload_max_filesize:
capture decran 2014-07-24 a 17 21 42 2

Screenshot of wrong mimetypes:
capture decran 2014-07-24 a 17 22 03 2

### CODE REVIEW

**Comment by Filoz** [comment]
This is really useful! :+1:

File: Provider/BaseProvider.php [hunk 1]
```diff
@@ -71,7 +71,10 @@ public function __construct($name, Filesystem $filesystem, CDNInterface $cdn, Ge
             return;
         }
 
-        $this->doTransform($media);
+        try {
+            $this->doTransform($media);
+        } catch (\Exception $e) {
+        }
```
Comment (rande) [inline Provider/BaseProvider.php]: Can you log the exception ?

**Comment by antoinemineau** [comment]
When the file is bigger than the upload_max_filesize, it returns this exception : https://github.com/antoinemineau/SonataMediaBundle/blob/c19bf05c328e1b0326ec6d0726bb74a071e65f02/Provider/FileProvider.php#L224

When the file has a wrong mimetypes it returns :
Notice: Undefined variable: image in /Applications/MAMP/htdocs/ideastim/vendor/sonata-project/media-bundle/Provider/ImageProvider.php line 105

Not sure why it returns this undefined variable. Instead of throwing exception, we could use a return to stop the function call and remove the try/catch from BaseProvider.

**Comment by borNfreee** [comment]
any news?

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**